OperacoesListaSequencial.cpp

```diff
@@ -0,0 +1,459 @@
+  *tamanhoDaListaSequencial = *tamanhoDaListaSequencial + 1;
+}
+
+// Remove o elemento inicial
+void removeInicioSequencial(pessoa*& ponteiroSequencial,
+                            int* tamanhoDaListaSequencial) {
+  // Cria um vetor com uma posicao a menos
+  pessoa* copiaListaSequencial = new pessoa[*tamanhoDaListaSequencial - 1];
+
+  // Passa os elementos da lista antiga ignorando o primeiro para a nova ate o
+  // elemento desejado
+  int cont;
+  for (cont = 1; cont < *tamanhoDaListaSequencial; cont++) {
+    copiaListaSequencial[cont - 1].nome = ponteiroSequencial[cont].nome;
+    copiaListaSequencial[cont - 1].rg = ponteiroSequencial[cont].rg;
+  }
+
+  // Faz o ponteiroSequencial apontar para a nova lista com um elemento extra
+  ponteiroSequencial = copiaListaSequencial;
+
+  // Atualiza o tamanho da lista
+  *tamanhoDaListaSequencial = *tamanhoDaListaSequencial - 1;
+}
+
+// Remove o elemento final
+void removeFimSequencial(pessoa*& ponteiroSequencial,
+                         int* tamanhoDaListaSequencial,
+                         int* posicaoNaLista) {
+  // Posicao na lista
+  *posicaoNaLista = *tamanhoDaListaSequencial - 1;
+
+  // Cria um vetor com uma posicao a menos
+  pessoa* copiaListaSequencial = new pessoa[*tamanhoDaListaSequencial - 1];
+
+  // Passa os elementos da lista antiga ignorando o primeiro para a nova ate o
+  // elemento desejado
+  int cont;
+  for (cont = 0; cont < *tamanhoDaListaSequencial - 1; cont++) {
+    copiaListaSequencial[cont].nome = ponteiroSequencial[cont].nome;
+    copiaListaSequencial[cont].rg = ponteiroSequencial[cont].rg;
+  }
+
+  // Faz o ponteiroSequencial apontar para a nova lista com um elemento extra
+  ponteiroSequencial = copiaListaSequencial;
+
+  // Atualiza o tamanho da lista
+  *tamanhoDaListaSequencial = *tamanhoDaListaSequencial - 1;
+}
+
+// Remove um elemento por posicao
+void removePosicaoSequencial(pessoa*& ponteiroSequencial,
+                             int* tamanhoDaListaSequencial,
+                             int posicao) {
+  // Cria um vetor com uma posicao a mais
+  pessoa* copiaListaSequencial = new pessoa[*tamanhoDaListaSequencial - 1];
+
+  // Passa os elementos da lista antiga para a nova ate o elemento desejado
+  int cont;
+  for (cont = 0; cont < *tamanhoDaListaSequencial - 1; cont++) {
+    if (cont < posicao) {
+      // Faz uma copia dos elementos ate a posicao
+      copiaListaSequencial[cont].nome = ponteiroSequencial[cont].nome;
+      copiaListaSequencial[cont].rg = ponteiroSequencial[cont].rg;
+    } else {
+      // Faz uma copia dos elementos ate a posicao
+      copiaListaSequencial[cont].nome = ponteiroSequencial[cont + 1].nome;
+      copiaListaSequencial[cont].rg = ponteiroSequencial[cont + 1].rg;
+    }
+  }
+
+  // Faz o ponteiroSequencial apontar para a nova lista com um elemento extra
+  ponteiroSequencial = copiaListaSequencial;
+  // Atualiza o tamanho da lista
+  *tamanhoDaListaSequencial = *tamanhoDaListaSequencial - 1;
+}
```

Comment:
this is implemented in a very memory inefficient way.
study this link https://www.bogotobogo.com/cplusplus/linkedlist.php and reimplement